web/features/search/SearchContext.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  useCallback,
+} from "react";
+import { getClient } from "../../utils/sanity/sanity.server";
+
+export type SearchResultType = {
+  slug: {
+    current: string;
+  };
+  title: string;
+  availableFrom: string;
+};
+
+const SearchContext = createContext(null);
+
+type SearchContextType = {
+  searchStr: string;
+  setSearchStr: (searchStr: string) => void;
+  loading: boolean;
+  searchResults: SearchResultType[];
+  searchIsActive: boolean;
+  search: () => void;
+  closeSearch: () => void;
```

Comment:
navngivningspirk igjen 👋🏻  Jeg foretrekker at alle booleans starter med `is`/`are` og at callbacks starter med `on`.

Så i denne konteksten:

```
loading -> isLoading
searchIsActive -> isSearchActive
search -> onSearch
closeSearch -> onClose
```

Reply:
Enig, det ser bedre ut :)